src/http-client/index.ts

```diff
@@ -14,9 +22,30 @@ abstract class HttpClient extends Dispatcher {
    */
   public static async request(options: HttpClientRequestOptions): Promise<Response> {
     let res = await this._request(options, false);
+    let txnAuthJSON: TxnAuthJSON | undefined;
+
     if (this.newTokenRequired(res, options.requiresNewToken)) {
       res = await this._request(options, true);
     }
+
+    if (options.txnAuth && options.txnAuth.init) {
+      if (res.redirected && (await examineForIGTxnAuth(res))) {
+        txnAuthJSON = await normalizeIGJSON(res);
+      } else if (await examineForRESTTxnAuth(res)) {
+        txnAuthJSON = await normalizeRESTJSON(res);
+      }
+
+      if (txnAuthJSON && txnAuthJSON.advices) {
+        const { serverConfig } = Config.get(options.txnAuth.options);
+        const txnAuthOptions = buildTxnAuthReqOptions(
+          txnAuthJSON,
+          serverConfig.baseUrl,
+          options.timeout,
+        );
+        res = await this._request(txnAuthOptions, false);
+      }
+    }
```

Comment:
In order to keep this code readable, understandable and maintainable, I decided to go with very small, focused functions that enable good semantics when read at this level as well as consistent ergonomics. It does have the downside of running a bit redundant within the units themselves.

If we prioritize efficiency more than simple readability, I can redesign the implementation. But, after many attempts at being both efficient and readable that didn't work out, I landed on prioritizing readability. Let me know if this is not desired, and I'll be happy to revisit how this is designed.